Change firmware version for Ubiquiti WA devices to 8.7.4.
Index: firmware/openwrt/target/linux/ath79/image/common-ubnt.mk
===================================================================
--- firmware.orig/openwrt/target/linux/ath79/image/common-ubnt.mk
+++ firmware/openwrt/target/linux/ath79/image/common-ubnt.mk
@@ -84,7 +84,7 @@ define Device/ubnt-wa
   UBNT_BOARD := WA
   UBNT_CHIP := ar934x
   UBNT_TYPE := WA
-  UBNT_VERSION := 8.5.3
+  UBNT_VERSION := 8.7.4
 endef
 
 define Device/ubnt-xc
