Allow optional 'chanlist' restriction for hostapd.

Signed-off-by: Lars Kruse <devel@sumpfralle.de>

--- b/openwrt/package/network/services/hostapd/files/netifd.sh	2015-03-25 02:26:23.000000000 +0100
+++ a/openwrt/package/network/services/hostapd/files/netifd.sh	2015-03-25 03:17:10.000000000 +0100
@@ -104,6 +104,7 @@
 
 	config_add_int maxassoc max_inactivity
 	config_add_boolean disassoc_low_ack isolate short_preamble
+	config_add_string chanlist
 
 	config_add_int \
 		wep_rekey eap_reauth_period \
@@ -161,7 +162,7 @@
 		maxassoc max_inactivity disassoc_low_ack isolate auth_cache \
 		wps_pushbutton wps_label ext_registrar wps_pbc_in_m1 \
 		wps_device_type wps_device_name wps_manufacturer wps_pin \
-		macfilter ssid wmm hidden short_preamble rsn_preauth
+		macfilter ssid wmm hidden short_preamble rsn_preauth chanlist
 
 	set_default isolate 0
 	set_default maxassoc 0
@@ -181,6 +182,9 @@
 	if [ "$max_inactivity" -gt 0 ]; then
 		append bss_conf "ap_max_inactivity=$max_inactivity" "$N"
 	fi
+	if [ -n "$chanlist" ]; then
+		append bss_conf "chanlist=$chanlist" "$N"
+	fi
 
 	append bss_conf "disassoc_low_ack=$disassoc_low_ack" "$N"
 	append bss_conf "preamble=$short_preamble" "$N"
