Allow optional 'chanlist' restriction for hostapd.

Signed-off-by: Lars Kruse <devel@sumpfralle.de>

Index: on_firmware/openwrt/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
===================================================================
--- on_firmware.orig/openwrt/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
+++ on_firmware/openwrt/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
@@ -28,6 +28,7 @@ drv_mac80211_init_device_config() {
 	config_add_string path phy 'macaddr:macaddr'
 	config_add_string hwmode
 	config_add_string tx_burst
+	config_add_string chanlist
 	config_add_string distance
 	config_add_int beacon_int chanbw frag rts
 	config_add_int rxantenna txantenna antenna_gain txpower
@@ -350,6 +351,10 @@ mac80211_hostapd_setup_base() {
 		[ -n "$vht_capab" ] && append base_cfg "vht_capab=$vht_capab" "$N"
 	fi
 
+	# define limited list of allowed channels (e.g. only outdoor channels)
+	json_get_vars chanlist
+	[ -n "$chanlist" ] && append base_cfg "chanlist=$chanlist" "$N"
+
 	hostapd_prepare_device_config "$hostapd_conf_file" nl80211
 	cat >> "$hostapd_conf_file" <<EOF
 ${channel:+channel=$channel}
