Show "client isolation" setting even for non-madwifi drivers.
--- a/luci/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua
+++ b/luci/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua
@@ -518,6 +518,10 @@ if hwtype == "mac80211" then
 	wmm:depends({mode="ap"})
 	wmm:depends({mode="ap-wds"})
 	wmm.default = wmm.enabled
+
+	isolate = s:taboption("general", Flag, "isolate", translate("Separate Clients"),
+	  translate("Prevents client-to-client communication"))
+	isolate:depends({mode="ap"})
	
	ifname = s:taboption("advanced", Value, "ifname", translate("Interface name"), translate("Override default interface name"))
	ifname.optional = true
 end
 
 
