die opennet-Pakete verwenden gemeinsame Makefile-Elemente
Index: on_firmware/openwrt/include/opennet.mk
===================================================================
--- /dev/null
+++ on_firmware/openwrt/include/opennet.mk
@@ -0,0 +1,54 @@
+# wir lesen die Versionsnummer aus der config-Datei
+VERSION_NUMBER = $(shell cat "$(TOPDIR)/.config" | grep "^CONFIG_VERSION_NUMBER=" | cut -f 2- -d = | tr -d \"\')
+ifeq "$(VERSION_NUMBER)" ""
+        # Wenn es noch keine CONFIG_VERSION_NUMBER in .config gibt,
+        # setze einen default Wert, um Fehlermeldungen zu vermeiden
+        VERSION_NUMBER="0.0.0-0"
+endif
+
+# Wir verwenden die Version des Gesamt-Builds
+# Wir teilen die VERSION_NUMBER (z.B. "0.5.0-694" oder "0.5.1-unstable-810") an der Stelle des letzten Bindestrichs
+PKG_VERSION = $(shell echo "$(VERSION_NUMBER)" | sed 's/-[0-9]\+\(-[0-9a-f]\+\)\?$$//g')
+PKG_RELEASE = $(shell echo "$(VERSION_NUMBER)" | sed 's/^.*-\([0-9]\+\(-[0-9a-f]\+\)\?\)$$/\1/g')
+PKG_BUILD_DIR := $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+PKG_BUILD_DEPENDS := $(if $(STAGING_DIR_ROOT),lua/host)
+
+# "package.mk" muss nach dem Setzen von PKG_VERSION und PKG_RELEASE geladen werden
+include $(INCLUDE_DIR)/package.mk
+
+
+PO_CONV = ../../../luci/modules/luci-base/src/po2lmo
+
+
+define Package/opennet/install/template
+	$(INSTALL_DIR) "$(1)/"
+	@# das on-goodies-Paket enthaelt keinerlei Dateien - also darf die Kopie fehlschlagen
+	$(CP) "$(PKG_BUILD_DIR)/files/." "$(1)/" || true
+	@# Zertifikate hashen, falls noetig (nur fuer on-certificates relevant; notwendig fuer csr-Upload via https)
+	[ -d "$(1)/etc/ssl/certs/opennet-initiative.de" ] && c_rehash "$(1)/etc/ssl/certs/opennet-initiative.de" || true
+	@# falls es keine ipkg-Dateien gibt, existiert das Quell-Verzeichnis in git nicht
+	$(CP) -L "$(PKG_BUILD_DIR)/ipkg/." "$(1)/CONTROL" || true
+	$(INSTALL_DIR) "$(1)/usr/lib/lua/luci/i18n"
+	# leere po-Dateien fuehren zu fehlenden lmo-Dateien (z.B. on-core) - das koennen wir ignorieren
+	$(CP) "$(PKG_BUILD_DIR)/lmo"/*.lmo "$(1)/usr/lib/lua/luci/i18n/" || true
+endef
+
+# falls Build/Compile nicht definiert wird, dann wird das Makefile aus dem build_dir geloescht
+# siehe https://forum.openwrt.org/viewtopic.php?pid=143534#p143534
+define Build/Compile
+endef
+
+define Build/Prepare
+	mkdir -p $(PKG_BUILD_DIR)/lmo; $(TAR) c \
+		--exclude=.pc --exclude=.svn --exclude=.git \
+		--exclude='boa-0*' --exclude='*.o' --exclude='*.so' \
+		--exclude=dist . | \
+			$(TAR) x -C $(PKG_BUILD_DIR)/
+	@# po2lmo bauen, falls nicht vorhanden
+	[ -x "$(PO_CONV)" ] || make -C ../../../luci/modules/luci-base/src/ po2lmo
+	@# "find" liefert leider keine Fehlermeldung, falls "-exec" fehlschlaegt - also loesen wir es manuell aus
+	[ -x "$(PO_CONV)" ] || { echo >&2 "ERROR: po2lmo does not exist"; exit 1; }
+	@# TODO: mehrere Sprachen?
+	find ../../po/de -type f -name "$(PKG_NAME).po" -exec "$(PO_CONV)" '{}' "$(PKG_BUILD_DIR)/lmo/$(PKG_NAME).de.lmo" \;
+endef
+
