Switch to newer v0.15.1 release of olsrd2
Index: on_firmware/routing/oonf-olsrd2/Makefile
===================================================================
--- on_firmware.orig/routing/oonf-olsrd2/Makefile
+++ on_firmware/routing/oonf-olsrd2/Makefile
@@ -1,8 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=oonf-olsrd2
-PKG_VERSION:=0.14.1-1
-PKG_REV:=d86825c0bdf144db706da444be7b0a2e8722b54d
+PKG_VERSION:=0.15.1
+PKG_REV:=bffb88b040659b237c4c91b6b42dbbb47431750e
 PKG_RELEASE:=$(PKG_REV)
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
