Notwendige Anpassung des Makefile fuer die hinzugefuegten ondataservice-Plugins

diff --git a/opennet/packages/olsrd/Makefile b/opennet/packages/olsrd/Makefile
index e176185..41ec140 100644
--- a/opennet/packages/olsrd/Makefile
+++ b/opennet/packages/olsrd/Makefile
@@ -94,6 +94,18 @@ define Package/olsrd-mod-nameservice
   TITLE:=Lightweight hostname resolver plugin
 endef
 
+define Package/olsrd-mod-ondataservice
+  $(call Package/olsrd/template)
+  DEPENDS:=olsrd +libsqlite3 +sqlite3-cli
+  TITLE:=Opennet Dataservice SQLite3 plugin
+endef
+
+define Package/olsrd-mod-ondataservice_light
+  $(call Package/olsrd/template)
+  DEPENDS:=olsrd
+  TITLE:=Opennet Dataservice JSON plugin
+endef
+
 define Package/olsrd-mod-p2pd
   $(call Package/olsrd/template)
   DEPENDS:=olsrd
@@ -147,7 +159,7 @@ MAKE_FLAGS+= \
 	DESTDIR="$(PKG_INSTALL_DIR)" \
 	STRIP="true" \
 	INSTALL_LIB="true" \
-	SUBDIRS="arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns nameservice p2pd pgraph quagga secure tas txtinfo watchdog"
+	SUBDIRS="arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns nameservice ondataservice ondataservice_light p2pd pgraph pud quagga secure sgwdynspeed txtinfo watchdog"
 
 define Build/Compile
 	$(call Build/Compile/Default,all)
@@ -208,6 +220,16 @@ define Package/olsrd-mod-nameservice/install
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/nameservice/olsrd_nameservice.so.* $(1)/usr/lib/
 endef
 
+define Package/olsrd-mod-ondataservice/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/ondataservice/olsrd_ondataservice.so.* $(1)/usr/lib/
+endef
+
+define Package/olsrd-mod-ondataservice_light/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/ondataservice_light/olsrd_ondataservice_light.so.* $(1)/usr/lib/
+endef
+
 define Package/olsrd-mod-p2pd/install
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/p2pd/olsrd_p2pd.so.* $(1)/usr/lib/
@@ -255,6 +277,8 @@ $(eval $(call BuildPackage,olsrd-mod-httpinfo))
 $(eval $(call BuildPackage,olsrd-mod-jsoninfo))
 $(eval $(call BuildPackage,olsrd-mod-mdns))
 $(eval $(call BuildPackage,olsrd-mod-nameservice))
+$(eval $(call BuildPackage,olsrd-mod-ondataservice))
+$(eval $(call BuildPackage,olsrd-mod-ondataservice_light))
 $(eval $(call BuildPackage,olsrd-mod-p2pd))
 $(eval $(call BuildPackage,olsrd-mod-pgraph))
 $(eval $(call BuildPackage,olsrd-mod-quagga))
